### DISCUSSION TITLE
Skywalking Add New Functionality How to test it without Building

### DISCUSSION THREAD

**Discussion by AjinThankachan** [apache/skywalking#9259] [Q&A]
Hi Team,
I am trying to add changes and some plugins to Skywalking and debug it. But it is taking too much time to Build everything. So is there any way via ecclipse or any other IDE we can put the jar and do the Realtime testing and after that build it. For a small change its taking 20 mins to build. Can someone help me. Thanks in advance.

**Comment by wu-sheng** [comment]
Are you asking about Java agent?

**Reply by AjinThankachan** [reply]
Sorry my bad I am using Skywalking 8.9.1 collector. I am doing change in that.

**Reply by wu-sheng** [reply]
Then, I don't know what you mean by building everything. It may be some issues with eclipse or how you use it? I don't use Eclipse for years.
In my IntelliJ case, I just change codes and reboot.

**Reply by AjinThankachan** [reply]
I have downloaded Intellije. So there i can directly compile that way you are saying. Instead of Building. I thought i need to do maven to compile the files also.

**Reply by wu-sheng** [reply]
Still confused what do you mean.

**Reply by AjinThankachan** [reply]
Usually when done change for eg: adding the change in the login page which is there in the code. I have uncommented the code and done  changes and added some validation also. Then we will build it using maven and create the tar zip file and extract it and run it in local and test whether its working or not. This is the how we are doing. 

 Insted of that what i wanted and i need is to made the changes and put the file/jar in the same and from there just run the IDE compile or RUN option. So i will be able to see the UI and test my login was working or not like localhosting. Then i can stop the host/code and try making things right. Thats what i was expecting. Hope now its clear for you. Just need to test the changes without builing the whole thing again and again. If you can share the doc or the method how you guys are making changes in code and testing that is also fine for me.

**Reply by wu-sheng** [reply]
You are working in a private development environment, which is far away from SkyWalking's committers' doing.
We don't develop features like you did. As a new feature, such as login, it relates to GraphQL protocol, backend implementation, UI development, etc. There is not a work for a single person. Every feature in the open-source community, it takes weeks async. 
GraphQL gets proposed, reviewed, and merged first, then backend and UI only need to work protocol-oriented.

What you are asking is not suitable for us actually.

**Reply by wu-sheng** [reply]
Open source is an async and remote collaboration. There is multi-people involved, and protocol-oriented. It is very rare to things like you mentioned.
